src/Coordination/FourLetterCommand.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+#include <sstream>
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+#include <Interpreters/Context_fwd.h>
+#include <IO/WriteBufferFromString.h>
+#include <common/types.h>
+
+namespace DB
+{
+struct FourLetterCommand;
+using FourLetterCommandPtr = std::shared_ptr<DB::FourLetterCommand>;
+
+struct FourLetterCommand
+{
+public:
+    using StringBuffer = DB::WriteBufferFromOwnString;
+    explicit FourLetterCommand(const Context & global_context_);
+
+    virtual String name() = 0;
+    virtual void run(String & res) = 0;
+
+    virtual ~FourLetterCommand();
+    Int32 code();
+
+    static void printSet(StringBuffer & ss, std::unordered_set<String> & target, String && prefix);
+
+protected:
+    const Context & global_context;
+};
+
+/***
+ * Tests if server is running in a non-error state. The server will respond with imok if it is running.
+ * Otherwise it will not respond at all.
+ *
+ * A response of "imok" does not necessarily indicate that the server has joined the quorum,
+ * just that the server process is active and bound to the specified client port.
+ * Use "stat" for details on state wrt quorum and client connection information.
+ */
+struct RuokCommand : public FourLetterCommand
+{
+    explicit RuokCommand(const Context & global_context_) : FourLetterCommand(global_context_) { }
+
+    String name() override;
+    void run(String & res) override;
+    ~RuokCommand() override;
+};
+
+struct FourLetterCommands
```

Comment:
It looks like factory. We have a lot of them in our code. Take a look at StorageFactory + registerStorages.